src/vm/database/sqlite.rs

```diff
@@ -221,6 +234,66 @@ impl <'a> ContractDatabase <'a> {
         }
     }
 
+    fn load_variable(&self, contract_name: &str, variable_name: &str) -> Result<SqliteDataVariable> {
+        let (variable_identifier, value_type): (_, String) =
+            self.query_row(
+                "SELECT variable_identifier, value_type FROM variables_table WHERE contract_name = ? AND variable_name = ?",
+                &[contract_name, variable_name],
+                |row| {
+                    (row.get(0), row.get(1))
+                })
+            .ok_or(UncheckedError::UndefinedVariable(variable_name.to_string()))?;
+
+        Ok(SqliteDataVariable {
+            variable_identifier: variable_identifier,
+            value_type: TypeSignature::deserialize(&value_type)
+        })
+    }
+
+    pub fn create_variable(&mut self, contract_name: &str, variable_name: &str, value_type: TypeSignature) {
+        self.execute("INSERT INTO variables_table (contract_name, variable_name, value_type) VALUES (?, ?, ?)",
+                     &[contract_name, variable_name, &value_type.serialize()]);
+    }
+
+    pub fn set_variable(&mut self, contract_name: &str, variable_name: &str, value: Value) -> Result<Value> {
+        let variable_descriptor = self.load_variable(contract_name, variable_name)?;
+        if !variable_descriptor.value_type.admits(&value) {
+            return Err(UncheckedError::TypeError(format!("{:?}", variable_descriptor.value_type), value).into())
+        }
+
+        let params: [&ToSql; 2] = [&variable_descriptor.variable_identifier,
+                                   &value.serialize()];
+
+        self.execute(
+            "INSERT INTO data_table (variable_identifier, value, key, map_identifier) VALUES (?, ?, '', '')",
```

Comment:
`map_identifier` is an integer.  Why is it being set to `''`, and how is this interpreted?  Why not `NULL`?

Reply:
'' was (abusevely) working. Updated with a `0` instead, thanks.
There is `NOT NULL` constraint that @kantai wants to keep.

Reply:
Addressed with https://github.com/blockstack/blockstack-core/pull/1022/commits/ead27b97d04ae71d521817d14eea84bc04823e11

Reply:
Hmm -- with a little more thought, I think allowing NULLs here would probably be the safer approach. `0` could be a legal `map_identifier`.